spring-session/src/main/java/org/springframework/session/web/http/SessionRepositoryFilter.java

```diff
@@ -132,6 +136,22 @@ public void setServletContext(ServletContext servletContext) {
 		this.servletContext = servletContext;
 	}
 
+	private final class HttpRequestSessionContext implements SessionContext {
```

Comment:
As far as I can see this class could be `static`, so it won't keep to its outer. This is preferable most of the time.